bioio_ome_zarr/writers/ome_zarr_writer_v3.py

```diff
@@ -0,0 +1,541 @@
+        self.num_levels = len(self.level_shapes)
+
+        # 5) Prepare chunk & shard parameters per level
+        self.chunks = self._prepare_parameter(chunks, self._suggest_chunks, "chunks")
+        self.shards = self._prepare_parameter(
+            shards, lambda s: s, "shards", required=False
+        )
+
+        # 6) Initialize Zarr store and create arrays for each level
+        self.root = self._init_store(store)
+        self.datasets = self._create_arrays(compressor)
+
+        # 7) Convert Channel objects to raw dicts (if provided)
+        if channels is not None:
+            self.channels = [ch.to_dict() for ch in channels]
+        else:
+            self.channels: Optional[List[Dict[str, Any]]] = None  # type: ignore
+
+        # 8) Store OMERO render settings verbatim
+        self.rdefs = rdefs
+
+        # 9) Capture optional top-level scale transform
+        self.multiscale_scale = multiscale_scale
+
+        # 10) Build axes metadata list
+        axes_meta = self.axes.to_metadata()
+
+        # 11) Write the OME-Zarr metadata block
+        self._write_metadata(
+            name=image_name,
+            axes=axes_meta,
+            channels=self.channels,
+            rdefs=self.rdefs,
+            creator=creator_info,
+        )
+
+    def _compute_levels(self, max_levels: Optional[int]) -> List[Tuple[int, ...]]:
+        """
+        Compute pyramid level shapes by repeatedly downsampling spatial axes.
+
+        Parameters
+        ----------
+        max_levels : Optional[int]
+            Maximum number of levels (including base). If None, generate
+            until downsampled shape equals previous.
+
+        Returns
+        -------
+        List[Tuple[int, ...]]
+            List of shape tuples for each level.
+
+        Notes
+        -----
+        * Only axes named "x" or "y" get downsampled by `scale_factors`.
+        * Stops early if new shape equals the last shape.
+        """
+        shapes: List[Tuple[int, ...]] = [self.shape]
+        lvl = 1
+
+        while max_levels is None or lvl < max_levels:
+            prev = shapes[-1]
+            nxt: List[int] = []
+            for i, size in enumerate(prev):
+                ax_name = self.axes.names[i].lower()
+                factor = self.axes.factors[i]
+                if ax_name in ("x", "y") and factor > 1:
+                    nxt.append(max(1, size // factor))
+                else:
+                    nxt.append(size)
+            nxt_tuple = tuple(nxt)
+            if nxt_tuple == prev:
+                break
+            shapes.append(nxt_tuple)
+            lvl += 1
+
+        return shapes
+
+    def _prepare_parameter(
+        self,
+        param: Any,
+        default_fn: Callable[[Tuple[int, ...]], Tuple[int, ...]],
+        name: str,
+        required: bool = True,
+    ) -> List[Optional[Tuple[int, ...]]]:
+        """
+        Standardize chunk or shard specification across levels.
```

Comment:
what does standardize mean?  I can't really tell what this function is supposed to do.  It takes a param (of type Any?  prob could type it more precisely) and a default_fn and returns what?    Looks like `name` could be typed more precisely too if it only accepts two possible values

Edit: I see that the "Notes" section sort of tells me what the function does.  But all my above comments are still valid, it's kinda hard to read this func and it looks like multiple functions piled into one.